lib/legacy/zstd_v02.c

```diff
@@ -3312,37 +3314,57 @@ static size_t ZSTD_decompress(void* dst, size_t maxDstSize, const void* src, siz
     return ZSTD_decompressDCtx(&ctx, dst, maxDstSize, src, srcSize);
 }
 
-static size_t ZSTD_findFrameCompressedSize(const void *src, size_t srcSize)
+MEM_STATIC void ZSTD_errorFrameSizeInfoLegacy(size_t* cSize, unsigned long long* dBound, size_t ret)
```

Comment:
nit: This function only needs to be `static`, not `MEM_STATIC`. That macro just tells the compiler it is static and it may be unused, but this function isn't unused. Same for the rest of the implementations of `ZSTD_errorFrameSizeInfoLegacy()`.

Reply:
Should I also add error checking? For instance:

```
if (cSize == NULL || dBound == NULL) return;
```

Reply:
I would just `assert(cSize != NULL && dBound != NULL)` at the top of each `ZSTDv0X_findFrameInfoLegacy()`. Since these functions are internal only, and not exposed (no-one includes `zstd_v0X.h` directly except `zstd_legacy.h`) we can assume that they get valid parameters, but adding an assert helps verify that during testing.

Reply:
Actually, these files don't have any asserts. I would just write in the docs that neither may be `NULL`. Since it is a `NULL` check debug builds would crash anyways.